src/qml/modules/Shotcut/Controls/GradientMapControl.qml

```diff
@@ -0,0 +1,307 @@
+
+            function getPosition() {
+                return (x + width / 2) / parent.width;
+            }
+
+            x: (typeof stopList[stopIndex] != 'undefined' ? stopList[stopIndex].position : 0.0) * parent.width - width / 2
+            y: 0
+            color: typeof stopList[stopIndex] != 'undefined' ? stopList[stopIndex].color : "gray"
+            border.color: "gray"
+            border.width: 1
+            width: 10
+            height: parent.height
+            radius: 2
+            visible: stopList.length > 1
+
+            onXChanged: {
+                if (stopArea.drag.active) {
+                    var position = getPosition();
+                    gradientMap._setStopPosition(stopIndex, position);
+                    stopPosition.value = position;
+                }
+            }
+
+            Shotcut.ColorDialog {
+                id: colorDialog
+
+                title: qsTr("Color #%1").arg(stopIndex + 1)
+                selectedColor: handleRect.color
+                onAccepted: gradientMap._setStopColor(handleRect.stopIndex, String(selectedColor), getPosition())
+            }
+
+            Shotcut.HoverTip {
+                text: qsTr('Color: %1\nClick to select, drag to change position').arg(color)
+            }
+
+            MouseArea {
+                id: stopArea
+                anchors.fill: parent
+                drag.target: parent
+                drag.axis: Drag.XAxis
+                drag.minimumX: ((parent.prev && parent.prev.position) || 0) * gradientFrame.width - width / 2
+                drag.maximumX: ((parent.next && parent.next.position) || 1) * gradientFrame.width - width / 2
+                drag.onActiveChanged: {
+                    if (!drag.active) {
+                        const position = getPosition();
+                        stopPosition.value = position;
+                        gradientMap._setStopColor(handleRect.stopIndex, String(handleRect.color), position);
+                    } else if (stopSpinner.value !== handleRect.stopIndex) {
+                        stopSpinner.value = handleRect.stopIndex;
+                    }
+                }
+                onClicked: {
+                    if (stopSpinner.value !== handleRect.stopIndex) {
+                        stopSpinner.value = handleRect.stopIndex;
+                    }
+                }
+            }
+        }
+    }
+
+    Rectangle {
+        id: gradientFrame
+
+        Layout.columnSpan: 3
+        Layout.fillWidth: true
+        implicitHeight: 20
+        implicitWidth: 200
+        color: "transparent"
+
+        Rectangle {
+            id: gradientRect
+
+            width: parent.width
+            height: parent.height - 6
+            border.color: "gray"
+            border.width: 1
+            radius: 4
+            gradient: Gradient {
+                id: gradientView
+                orientation: Gradient.Horizontal
+            }
+        }
+    }
+
+    Shotcut.DoubleSpinBox {
```

Comment:
This should have a label such as "Selected Stop". Also, it is cleaner if the UI is 1-based instead of 0-based.

Reply:
I'm kinda against the label to be honest, probably a hover tip would be better.

Reply:
Bare controls with no label and numeric fields suffix have no meaning! It is like a puzzle game. Do you have a way to view the UI you created? Basically, you can extract the latest Shotcut portable build, replace the `libmltplus.so` that contains your new `gradientmap` filter and edit these QML files in `Shotcut.app/share/shotcut/qml/...`
Here is what it looks like now that is completely unacceptable for my standards with Shotcut
![image](https://github.com/user-attachments/assets/6d698a47-da7c-4670-8474-bd84d68f37d3)
The numeric fields are not layed out correctly or in any 
Why does the selected stop's color button need to be the full width of the dock?

Here, I fixed it for you.
![image](https://github.com/user-attachments/assets/c0234cf8-b381-4973-b111-e7ff2167e7b7)

Reply:
Ah, I've took out the width value of both spinboxes and they've done that weird squish overlap, that was an oversight from my part.

> Why does the selected stop's color button need to be the full width of the dock?

I thought that it was more intuitive for users that way, I'm more familiarized with Blender in that respect. As per what it stands now from your retouches it looks good.